core/src/server/host_filtering/host.rs

```diff
@@ -162,20 +119,42 @@ impl AllowHosts {
 	}
 }
 
+fn parse_host(input: &str) -> (String, Port) {
+	// Remove possible protocol definition such as `wss://<hostname>:<port>/<route>`.
+	let with_route = input.split("://").last().expect(SPLIT_PROOF);
+
+	// Remove everything beyond `/` which is part of the route such as `<hostname>:<port>/<route>`.
+	let mut it = with_route.split('/');
+	let host_and_port = it.next().expect(SPLIT_PROOF).to_lowercase();
+
+	// The rest is `host:<optional port>`.
+	let mut it = host_and_port.split(":");
```

Comment:
Just to check, is `username:password@hostname:port/route` possible also and should the parsing account for that?

Reply:
I think you are correct, I'll try to refactor this code and I never liked this and I think we are better off using `Uri::parse` here instead.